### PR TITLE
chore: update pull request template for contributors

### PR DESCRIPTION
**Pull request by majorgreys**
**PREVIEW**

## Checklist
- [ ] Fill out pull request sections for feature change.
- [ ] Ensure tests are passing for affected code.
- [ ] [Library documentation](https://github.com/DataDog/dd-trace-py/tree/1.x/docs) and/or [Datadog's documentation site](https://github.com/DataDog/documentation/) is updated. Link to doc PR in description.

## Checklist
- [ ] Fill out pull request sections for bug fix.
- [ ] Ensure tests are passing for affected code.
- [ ] [Library documentation](https://github.com/DataDog/dd-trace-py/tree/1.x/docs) and/or [Datadog's documentation site](https://github.com/DataDog/documentation/) is updated. Link to doc PR in description.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,7 +1,38 @@
+## Description
 <!-- Briefly describe the change and why it was required. -->
+<!-- If this is a breaking change, explain why it s necessary. Breaking changes must append `!` after the type/scope. See https://ddtrace.readthedocs.io/en/stable/contributing.html for more details. -->
```
Comment (Kyle-Verhoog) [inline .github/PULL_REQUEST_TEMPLATE.md:3]:
```suggestion
<!-- If this is a breaking change, explain why it is necessary. Breaking changes must append `!` after the type/scope. See https://ddtrace.readthedocs.io/en/stable/contributing.html for more details. -->
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,7 +1,38 @@
...
+<!-- Expand on why the change is requried, include relevant context for reviewers -->
+
+## Design 
+<!-- Include benefits from the change as well as possible drawbacks and trade-offs -->
+
+## Testing strategy
+<!-- Describe the automated tests and/or the steps for manual testing.
+
+<!-- END feat -->
+
```
Comment (Kyle-Verhoog) [inline .github/PULL_REQUEST_TEMPLATE.md:24]:
```suggestion


```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -1,7 +1,38 @@
...
+<!-- START feat -->
+
+## Checklist
+- [ ] Fill out pull request sections for feature change.
+- [ ] Ensure tests are passing for affected code.
+- [ ] [Library documentation](https://github.com/DataDog/dd-trace-py/tree/1.x/docs) and/or [Datadog's documentation site](https://github.com/DataDog/documentation/) is updated. Link to doc PR in description.
+
+## Motivation
+<!-- Expand on why the change is requried, include relevant context for reviewers -->
```
Comment (Kyle-Verhoog) [inline .github/PULL_REQUEST_TEMPLATE.md:15]:
```suggestion
<!-- Expand on why the change is required, include relevant context for reviewers -->
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -1,7 +1,38 @@
...
-- [ ] Added to the correct milestone.
-- [ ] Tests provided or description of manual testing performed is included in the code or PR.
-- [ ] Library documentation is updated.
-- [ ] [Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).
+- [ ] Fill out pull request sections for bug fix.
+- [ ] Ensure tests are passing for affected code.
+- [ ] [Library documentation](https://github.com/DataDog/dd-trace-py/tree/1.x/docs) and/or [Datadog's documentation site](https://github.com/DataDog/documentation/) is updated. Link to doc PR in description.
+
+## Relevant issue(s)
+<!-- Link the pull request to any issues related to the fix. Use keywords for links to automate closing the issues resolved once the pull request is merged.  See https://ddtrace.readthedocs.io/en/stable/contributing.html for more details. -->
```
Comment (Kyle-Verhoog) [inline .github/PULL_REQUEST_TEMPLATE.md:32]:
```suggestion
<!-- Link the pull request to any issues related to the fix. Use keywords for links to automate closing the issues once the pull request is merged. See https://ddtrace.readthedocs.io/en/stable/contributing.html for more details. -->
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -1,7 +1,39 @@
+## Description
 <!-- Briefly describe the change and why it was required. -->
+<!-- If this is a breaking change, explain why it is necessary. Breaking changes must append `!` after the type/scope. See https://ddtrace.readthedocs.io/en/stable/contributing.html for more details. -->
+
+<!-- Copy and paste the relevant snippet based on the type of pull request -->
```
Comment (brettlangdon) [inline .github/PULL_REQUEST_TEMPLATE.md:5]: Does this selection not work anymore? Did it not work for PRs?

Comment (brettlangdon) [reply .github/PULL_REQUEST_TEMPLATE.md:5]: Should we automate this checklist based on the conventional commit type?

Comment (Kyle-Verhoog) [reply .github/PULL_REQUEST_TEMPLATE.md:5]: the selection was only ever available via the github CLI 😕 

they don't have any UI for using/selecting a template

**Review by ZStriker19 — APPROVED** [review]
LGTM! I think what we ask for is logical, should be easy to fill out for contributors, and helpful for reviewers.

**Pull request merged**